### PR TITLE
Added size info to anal classes

### PR DESCRIPTION
**Pull request by HoundThe**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
I've added size member to a class vtable, added size as an optional parameter for "acv" command, started storing the size to the sdb and modified _r_anal_class_list_vtables()_ to load and print the size. Based on #12601

### CODE REVIEW

**Comment by XVilka** [comment]
Failed tests (see https://travis-ci.com/radareorg/radare2/jobs/293722218#L1098 )
```
[XX] db/cmd/cmd_ac ac-  289

$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-17162Si1NZHT9Pq2i -

ac Album

ac PeripheryAlbum

acb PeripheryAlbum Album

ac Flatline

acb Flatline PeripheryAlbum

acm Flatline PrayerPosition 0x1337 0x42

acv Flatline 0x1337

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

?e --

ac- Album

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

?e --

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

?e --

ac- Flatline

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

-Album

-Flatline: PeripheryAlbum

-  (vtable at 0x1337)

-  PrayerPosition @ 0x1337 (vtable + 0x42)

-PeripheryAlbum: Album

-Album=c

-Flatline=c

-PeripheryAlbum=c

--

-attr.Flatline.base=0

-attr.Flatline.base.0=PeripheryAlbum,0

-attr.Flatline.method=PrayerPosition

-attr.Flatline.method.PrayerPosition=4919,66

-attr.Flatline.vtable=0

-attr.Flatline.vtable.0=0x1337,0

-attr.PeripheryAlbum.base=0

-attr.PeripheryAlbum.base.0=Album,0

-attrtypes.Flatline=base,method,vtable

-attrtypes.PeripheryAlbum=base

-0x00001337 0 method.Flatline.PrayerPosition

-0x00001337 0 vtable.Flatline.0

---

-Flatline: PeripheryAlbum

-  (vtable at 0x1337)

-  PrayerPosition @ 0x1337 (vtable + 0x42)

-PeripheryAlbum

-Flatline=c

-PeripheryAlbum=c

--

-attr.Flatline.base=0

-attr.Flatline.base.0=PeripheryAlbum,0

-attr.Flatline.method=PrayerPosition

-attr.Flatline.method.PrayerPosition=4919,66

-attr.Flatline.vtable=0

-attr.Flatline.vtable.0=0x1337,0

-attrtypes.Flatline=base,method,vtable

-0x00001337 0 method.Flatline.PrayerPosition

-0x00001337 0 vtable.Flatline.0

---

-Flatline: PeripheryAlbum

-  (vtable at 0x1337)

-  PrayerPosition @ 0x1337 (vtable + 0x42)

-PeripheryAlbum

-Flatline=c

-PeripheryAlbum=c

--

-attr.Flatline.base=0

-attr.Flatline.base.0=PeripheryAlbum,0

-attr.Flatline.method=PrayerPosition

-attr.Flatline.method.PrayerPosition=4919,66

-attr.Flatline.vtable=0

-attr.Flatline.vtable.0=0x1337,0

-attrtypes.Flatline=base,method,vtable

-0x00001337 0 method.Flatline.PrayerPosition

-0x00001337 0 vtable.Flatline.0

---

-PeripheryAlbum

-PeripheryAlbum=c

--

EXPECT=''

[XX] db/cmd/cmd_ac acv  306

$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-171625SYALTrmhVZJ -

ac Deadwing

acv Deadwing 0x1337

acv Deadwing 0xdeadbeef 0x42

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

-Deadwing

-  (vtable at 0x1337)

-  (vtable at 0xdeadbeef in class at +0x42)

-Deadwing=c

--

-attr.Deadwing.vtable=0,1

-attr.Deadwing.vtable.0=0x1337,0

-attr.Deadwing.vtable.1=0xdeadbeef,66

-attrtypes.Deadwing=vtable

-0x00001337 0 vtable.Deadwing.0

-0xdeadbeef 0 vtable.Deadwing.1

EXPECT=''

[XX] db/cmd/cmd_ac acn  178

$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-171627leleyAWuJVB -

ac Album

ac PorcupineTreeAlbum

acb PorcupineTreeAlbum Album

ac Deadwing

acb Deadwing PorcupineTreeAlbum 0x8

acv PorcupineTreeAlbum 0x1337 0x42

acm PorcupineTreeAlbum Listen 0xdeadbeef

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

?e --

acn PorcupineTreeAlbum Masterpiece

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

 Album

 Deadwing: PorcupineTreeAlbum

 PorcupineTreeAlbum: Album

   (vtable at 0x1337 in class at +0x42)

   Listen @ 0xdeadbeef

 Album=c

 Deadwing=c

 PorcupineTreeAlbum=c

 -

 attr.Deadwing.base=0

 attr.Deadwing.base.0=PorcupineTreeAlbum,8

 attr.PorcupineTreeAlbum.base=0

 attr.PorcupineTreeAlbum.base.0=Album,0

 attr.PorcupineTreeAlbum.method=Listen

 attr.PorcupineTreeAlbum.method.Listen=3735928559,-1

 attr.PorcupineTreeAlbum.vtable=0

-attr.PorcupineTreeAlbum.vtable.0=0x1337,66

+attr.PorcupineTreeAlbum.vtable.0=0x1337,66,0

 attrtypes.Deadwing=base

 attrtypes.PorcupineTreeAlbum=base,vtable,method

 0x00001337 0 vtable.PorcupineTreeAlbum.0

 0xdeadbeef 0 method.PorcupineTreeAlbum.Listen

 --

 Album

 Deadwing: Masterpiece

 Masterpiece: Album

   (vtable at 0x1337 in class at +0x42)

   Listen @ 0xdeadbeef

 Album=c

 Deadwing=c

 Masterpiece=c

 -

 attr.Deadwing.base=0

 attr.Deadwing.base.0=Masterpiece,8

 attr.Masterpiece.base=0

 attr.Masterpiece.base.0=Album,0

 attr.Masterpiece.method=Listen

 attr.Masterpiece.method.Listen=3735928559,-1

 attr.Masterpiece.vtable=0

-attr.Masterpiece.vtable.0=0x1337,66

+attr.Masterpiece.vtable.0=0x1337,66,0

 attrtypes.Deadwing=base

 attrtypes.Masterpiece=base,vtable,method

 0x00001337 0 vtable.Masterpiece.0

 0xdeadbeef 0 method.Masterpiece.Listen

EXPECT='Album

Deadwing: PorcupineTreeAlbum

PorcupineTreeAlbum: Album

  (vtable at 0x1337 in class at +0x42)

  Listen @ 0xdeadbeef

Album=c

Deadwing=c

PorcupineTreeAlbum=c

-

attr.Deadwing.base=0

attr.Deadwing.base.0=PorcupineTreeAlbum,8

attr.PorcupineTreeAlbum.base=0

attr.PorcupineTreeAlbum.base.0=Album,0

attr.PorcupineTreeAlbum.method=Listen

attr.PorcupineTreeAlbum.method.Listen=3735928559,-1

attr.PorcupineTreeAlbum.vtable=0

attr.PorcupineTreeAlbum.vtable.0=0x1337,66,0

attrtypes.Deadwing=base

attrtypes.PorcupineTreeAlbum=base,vtable,method

0x00001337 0 vtable.PorcupineTreeAlbum.0

0xdeadbeef 0 method.PorcupineTreeAlbum.Listen

--

Album

Deadwing: Masterpiece

Masterpiece: Album

  (vtable at 0x1337 in class at +0x42)

  Listen @ 0xdeadbeef

Album=c

Deadwing=c

Masterpiece=c

-

attr.Deadwing.base=0

attr.Deadwing.base.0=Masterpiece,8

attr.Masterpiece.base=0

attr.Masterpiece.base.0=Album,0

attr.Masterpiece.method=Listen

attr.Masterpiece.method.Listen=3735928559,-1

attr.Masterpiece.vtable=0

attr.Masterpiece.vtable.0=0x1337,66,0

attrtypes.Deadwing=base

attrtypes.Masterpiece=base,vtable,method

0x00001337 0 vtable.Masterpiece.0

0xdeadbeef 0 method.Masterpiece.Listen'

[XX] db/cmd/cmd_ac acv-  308

$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-17162D2CbT5e2y2Ou -

ac Deadwing

acv Deadwing 0x1337

acv Deadwing 0xdeadbeef 0x42

acll

k anal/classes/*

?e -

k anal/classes/attrs/*

f@F:classes

?e --

acv- Deadwing 0

acll

f@F:classes

-Deadwing

-  (vtable at 0x1337)

-  (vtable at 0xdeadbeef in class at +0x42)

-Deadwing=c

--

-attr.Deadwing.vtable=0,1

-attr.Deadwing.vtable.0=0x1337,0

-attr.Deadwing.vtable.1=0xdeadbeef,66

-attrtypes.Deadwing=vtable

-0x00001337 0 vtable.Deadwing.0

-0xdeadbeef 0 vtable.Deadwing.1

---

-Deadwing

-  (vtable at 0xdeadbeef in class at +0x42)

-0xdeadbeef 0 vtable.Deadwing.1

EXPECT=''
```

**Comment by HoundThe** [comment]
Yes, there was a segfault when only class_name argument was passed (the default case). Fixed that and now the travis is failing only because I changed printing (now vtable_list() prints the size aswell) and I added the size to the sdb, so dumping classes doesn't match the expected result.

**Comment by radare** [comment]
not green yet

Screenshot 2020-03-04 at 17 24 08

**Comment by HoundThe** [comment]
> not green yet
> 
> Screenshot 2020-03-04 at 17 24 08

Yes, that's because I added the size to the object, so it can't be green or am I missing something

**Comment by XVilka** [comment]
Looks green now, thanks!

File: libr/anal/class.c [hunk 1]
```diff
@@ -424,6 +424,13 @@ static void r_anal_class_set_flag(RAnal *anal, const char *name, ut64 addr) {
 	anal->flg_class_set (anal->flb.f, name, addr, 0);
 }
 
+static void r_anal_class_vtable_set_flag(RAnal *anal, const char *name, ut64 addr, ut32 size) {
```
Comment (thestr4ng3r) [inline libr/anal/class.c]: The meaning of this function isn't really specific to vtables. You can use it as a direct replacement of `r_anal_class_set_flag()` and just pass 0 as the size at the old callsites.

File: libr/include/r_anal.h [hunk 2]
```diff
@@ -1980,6 +1981,8 @@ R_API RAnalClassErr r_anal_class_vtable_delete(RAnal *anal, const char *class_na
 R_API void r_anal_class_list(RAnal *anal, int mode);
 R_API void r_anal_class_list_bases(RAnal *anal, const char *class_name);
 R_API void r_anal_class_list_vtables(RAnal *anal, const char *class_name);
+R_API void r_anal_class_list_offset_vfunction(RAnal *anal, const char *class_name, size_t offset);
```
Comment (thestr4ng3r) [inline libr/include/r_anal.h]: The definition of this function is not part of this pr, is it?

Comment (HoundThe) [reply libr/include/r_anal.h]: Yeah, you're right

Comment (HoundThe) [reply libr/include/r_anal.h]: Removed prototype

File: libr/anal/class.c (hunk 1 above)
Comment (HoundThe) [reply libr/anal/class.c]: Yes, I a changed back to using r_anal_class_set_flag()

**Comment by radare** [comment]
lgtm can you rebase to make it green?
